### PR TITLE
minor fix up and consistency edit for JsonSchema impls

### PR DESCRIPTION
**Pull request by ahl**
I took a pass through our `impl JsonSchema`s for consistency and some very small fix ups.

### CODE REVIEW

File: common/src/api/external/mod.rs [hunk 1]
```diff
@@ -200,19 +200,17 @@ where
     }
 }
 
-/// Custom JsonSchema implementation to encode the constraints on Name
-// TODO: 1. make this part of schemars w/ rename and maxlen annotations
-// TODO: 2. integrate the regex with `try_from`
```
Comment (ahl) [inline common/src/api/external/mod.rs:205]: we're not doing these things

File: common/src/api/external/mod.rs [hunk 2]
```diff
@@ -286,14 +283,13 @@ impl RoleName {
     }
 }
 
-/// Custom JsonSchema implementation to encode the constraints on Name
-// TODO see TODOs on Name above
+/// Custom JsonSchema implementation to encode the constraints on RoleName
```
Comment (ahl) [inline common/src/api/external/mod.rs:286]: fix copy/paste

File: common/src/api/external/mod.rs [hunk 3]
```diff
@@ -1063,30 +1071,34 @@ impl JsonSchema for Ipv6Net {
...
-                })),
-                instance_type: Some(schemars::schema::SingleOrVec::Single(Box::new(schemars::schema::InstanceType::String))),
-                string: Some(Box::new(schemars::schema::StringValidation {
-                    // Fully-specified IPv6 address. 4 hex chars per segment, 8 segments, 7
-                    // ":"-separators, slash and up to 3 subnet digits
-                    max_length: Some(43),
-                    min_length: None,
-                    pattern: Some(
-                        // Conforming to unique local addressing scheme, `fd00::/8`.
-                        concat!(
```
Comment (ahl) [inline common/src/api/external/mod.rs:1082]: this `concat!` wasn't doing anything

File: common/src/api/external/mod.rs [hunk 4]
```diff
@@ -1063,30 +1071,34 @@ impl JsonSchema for Ipv6Net {
...
-            }
-        )
+            })),
+            instance_type: Some(schemars::schema::InstanceType::String.into()),
+            string: Some(Box::new(schemars::schema::StringValidation {
+                pattern: Some(
+                    // Conforming to unique local addressing scheme,
+                    // `fd00::/8`.
+                    concat!(
+                        r#"^([fF][dD])[0-9a-fA-F]{2}:("#,
```
Comment (ahl) [inline common/src/api/external/mod.rs:1090]: the only change is `(fd|FD)` -> `[fF][dD]`

File: common/src/api/external/mod.rs [hunk 5]
```diff
@@ -1063,30 +1071,34 @@ impl JsonSchema for Ipv6Net {
...
-                    title: Some("An IPv6 subnet".to_string()),
-                    description: Some("An IPv6 subnet, including prefix and subnet mask".to_string()),
-                    examples: vec!["fd12:3456::/64".into()],
-                    ..Default::default()
-                })),
-                instance_type: Some(schemars::schema::SingleOrVec::Single(Box::new(schemars::schema::InstanceType::String))),
-                string: Some(Box::new(schemars::schema::StringValidation {
-                    // Fully-specified IPv6 address. 4 hex chars per segment, 8 segments, 7
-                    // ":"-separators, slash and up to 3 subnet digits
-                    max_length: Some(43),
```
Comment (ahl) [inline common/src/api/external/mod.rs:1078]: redundant w/ the regex

File: nexus/src/external_api/shared.rs [hunk 6]
```diff
@@ -182,25 +182,19 @@ impl JsonSchema for IpRange {
         gen: &mut schemars::gen::SchemaGenerator,
     ) -> schemars::schema::Schema {
         schemars::schema::SchemaObject {
-            metadata: Some(
-                schemars::schema::Metadata { ..Default::default() }.into(),
-            ),
```
Comment (ahl) [inline nexus/src/external_api/shared.rs:187]: that's not doing anything...

File: common/src/api/external/mod.rs [hunk 7]
```diff
@@ -200,19 +200,17 @@ where
...
     }
     fn json_schema(
-        _gen: &mut schemars::gen::SchemaGenerator,
+        _: &mut schemars::gen::SchemaGenerator,
     ) -> schemars::schema::Schema {
-        schemars::schema::Schema::Object(schemars::schema::SchemaObject {
+        schemars::schema::SchemaObject {
             metadata: Some(Box::new(schemars::schema::Metadata {
-                title: Some("A name used in the API".to_string()),
+                title: Some("A contextually unique name".to_string()),
```
Comment (ahl) [inline common/src/api/external/mod.rs:213]: @davepacheco do you think is a better or worse description?

Comment (davepacheco) [reply common/src/api/external/mod.rs:213]: Better!  "contextually" is a little vague though -- maybe "a name unique within the parent collection".  I was going to say "a user-controlled, mutable string identifier that is unique within the object's collection" but maybe those other details are better for the description.

Comment (ahl) [reply common/src/api/external/mod.rs:213]: I like it; thanks for the suggestion

**Pull request merged**